File: java-algorithm/src/main/java/com/leetcode/Problem344.java
package com.leetcode;

/**
 * 344. Reverse String
 * https://leetcode.com/problems/reverse-string/
 */
public class Problem344 {
    public void reverseString(char[] s) {
        int i = 0, j = s.length - 1;
        while (i < j) {
            char tmp = s[i];
            s[i++] = s[j];
            s[j--] = tmp;
        }
    }
}

File: python-algo/leetcode/problem_344.py
"""344. Reverse String
https://leetcode.com/problems/reverse-string/
"""
from typing import List


class Solution:
    def reverse_string(self, s: List[str]) -> None:
        i, j = 0, len(s) - 1
        while i < j:
            s[i], s[j] = s[j], s[i]
            i += 1
            j -= 1
